### PR TITLE
remove modal for inline edit category name, and autofocus

### LINKED ISSUE #930: Annotations: Inconsistency in method for editing main label name vs sub-label name
Editing the main label’s name pops up a modal with a blank field, but editing the sub-label keeps the original name intact, but allows user to edit it inline.

Recommend: make both editing methods consistent. I prefer the sub-label (inline) method as it’s less heavy-handed than a modal and also keeps the original name intact as a starting point (e.g. maybe you just wanted to fix a typo in the original name… this happens to me all the time).

### PR DESCRIPTION
**Pull request by colinmegill**
Fixes #930

### CODE REVIEW

File: client/src/components/categorical/category.js [hunk 1]
```diff
@@ -238,58 +238,66 @@ class Category extends React.Component {
               {""}
             </label>
             {/* Dialog uses portal, can be anywhere/factored out */}
```
Comment (bkmartinjr) [inline client/src/components/categorical/category.js]: What does this comment mean?

File: client/src/components/categorical/category.js [hunk 2]
```diff
@@ -238,58 +238,66 @@ class Category extends React.Component {
...
+                }
               }}
             >
               {isUserAnno ? (
                 <Icon style={{ marginRight: 5 }} icon="tag" iconSize={16} />
               ) : null}
-              {metadataField}
+
+              {annotations.isEditingCategoryName &&
+              annotations.categoryEditable === metadataField ? (
```
Comment (bkmartinjr) [inline client/src/components/categorical/category.js]: the name `categoryEditable` is confusing -- if I read the code right, it contains the name of the category currently being edited, ie, the category in "editable mode".    Consider renaming to `.categoryBeingEdited` or some such.

File: client/src/components/categorical/category.js [hunk 3]
```diff
@@ -238,58 +238,66 @@ class Category extends React.Component {
...
+
             <span
               data-testid={`category-expand-${metadataField}`}
               style={{
                 cursor: "pointer",
                 display: "inline-block"
               }}
               onClick={() => {
-                this.setState({ isExpanded: !isExpanded });
+                const editingCategory =
```
Comment (bkmartinjr) [inline client/src/components/categorical/category.js]: What prevents the user from trying to put two category names into edit mode?    A quick skim of the code makes it look possible -- `activateCategoryEditMode()` in `category.js` doesn't seem to check for this.

Comment (bkmartinjr) [reply client/src/components/categorical/category.js]: which leads to the question: does label editing state have the same issue?

Comment (colinmegill) [reply client/src/components/categorical/category.js]: Let's open a new issue for validation. Issue #930 just concerned changing the form of the input.

Comment (bkmartinjr) [reply client/src/components/categorical/category.js]: OK w/ me.  But please file it and cross-link to this PR so we don't lose track of it.

File: client/src/components/categorical/category.js (hunk 1 above)
Comment (colinmegill) [reply client/src/components/categorical/category.js]: See `portalContainer` here: https://blueprintjs.com/docs/#core/components/dialog

It's going to be attached to `document.body`. Similarly, we have componetized our `toasters` and called them in functions. Rendering is independent of the DOM structure, unlike all the other react elements in the component tree.

Comment (bkmartinjr) [reply client/src/components/categorical/category.js]: What I'm confused about is that the Dialog was removed by this PR.   So why is the comment still present in the code?

File: client/src/components/categorical/category.js (hunk 2 above)
Comment (colinmegill) [reply client/src/components/categorical/category.js]: Will change

**Comment by bkmartinjr** [comment]
also, please update the PR description to indicate it fixes 930.  TY

**Comment by colinmegill** [comment]
@bkmartinjr it already does — first line, right above the image :)

**Pull request merged**